addons/website/static/src/components/navbar/navbar.scss

```diff
@@ -0,0 +1,14 @@
+header {
+    .o_main_navbar {
+        .o_website_systray {
+            --NavBar-entry-margin-right: #{$o-navbar-entry-padding-h};
+        }
+
+        @if (not $o-navbar-use-pills) {
+            .o_website_systray .o_menu_systray_item > .btn {
+                /* This line is extracted from `%-main-navbar-entry-spacing` */
+                line-height: calc(var(--o-navbar-height) - #{$o-navbar-padding-v * 2});
```

Comment:
- I can't extend it, it comes with a lot of rules which we don't need sadly (and which breaks the layout)
- I don't think we should do anything in the web app, `o_menu_systray_item` is only website related? It was initially in the web app but @stefanorigano pointed out that it was a website only class, which I didn't realize, so I moved it to website with this [push force](https://github.com/odoo/odoo/compare/46b40997eaac9201acfedcfbbea4c1937d75ae63..8653f8f75dd6bde0d01a27443cdb6dbc3014c73f)

Reply:
Did not realize `o_menu_systray_item` was a website class either but my point was that the line-height should be set at framework level (it can be added on the navbar itself actually) and inherited by specific stuff (if website really need specific stuff).

Reply:
> I don't think we should do anything in the web app, o_menu_systray_item is only website related? It was initially in the web app but @stefanorigano pointed out that it was a website only class, which I didn't realize, so I moved it to website with this [push force](https://github.com/odoo/odoo/compare/46b40997eaac9201acfedcfbbea4c1937d75ae63..8653f8f75dd6bde0d01a27443cdb6dbc3014c73f)

@rdeodoo ,  there is a misunderstanding :)
`o_menu_systray_item` it's **not** website only... now I do understand your answer on discord ^^

To clarify what I meant:  if you experience misalignments in website only, then the fix should be placed there. Otherwise a general fix is needed ofc.

I've got in touch because we discussed about a website's issue (translate button etc..) and I was initially surprised to see your fix in the main app, I wanted to double-check (again, I can't reproduce the issue).


---


> Jesus, that line is ugly

@qsm-odoo , what's annoying you here 😅 ? 
```scss
 line-height: calc(var(--o-navbar-height) - #{$o-navbar-padding-v * 2});
 ```
`--o-navbar-height` adapts according to the screen size* and `$o-navbar-padding-v` according to the version  (community uses `0px`, enterprise uses `10px` instead).

(*) since the very last _milk_ iteration the navbar height it's fixed, thus we could replace the CSS variable for a SCSS one, meaning no need of `calc()` anymore. I suspect that's 'cause you're not a fan of `calc()` in general (and I understand why)... but probably we'll have to live with it until either CSS provides a "better" way to handle variables interpolation or sass allows us to define variables into media queries (probably just not possible).
There are workarounds I know, but this seemed the cleanest solution to me 🤷🏻‍♂️



> in standard web app, o_menu_systray_item should have the correct line-height, then at worst this should be about adding a line-height: inherit on the inner button here.

Well, `line-height` is not the only property we have to tickle in order to achieve consistency across all the different types of entries. I found cleaner to extend pseudo-classes rather that trying to force specific values for each element. Now that all the code moved to OWL (and we finally reached a good level of consistency) we could simplify I guess, I'll have a look when possible.

Reply:
@stefanorigano 

> To clarify what I meant: if you experience misalignments in website only, then the fix should be placed there. Otherwise a general fix is needed ofc.
> 
> I've got in touch because we discussed about a website's issue (translate button etc..) and I was initially surprised to see your fix in the main app, I wanted to double-check (again, I can't reproduce the issue).

Well, there is no case other than website for now, but if you create any `.o_menu_systray_item` element with a `.btn` inside in any systray in Odoo, it will have the same issue.
For instance, you can just copy this HTML inside the regular Odoo systray and see the vertical misalignment:
```
<a href="#" class="h-100 btn btn-primary d-flex align-items-center rounded-0 border-0 fw-normal" accesskey="a">
    <span class="fa me-2 fa-pencil"/>Edit
</a>
```
![image](https://github.com/odoo/odoo/assets/30048408/334f9a7e-c4e1-41ca-9760-d2b704f78207)

Reply:
> > Jesus, that line is ugly
>
> @qsm-odoo , what's annoying you here 😅 ?

@stefanorigano Nothing annoys me, I said it was ugly not useless :wink: You can be ugly and needed, look at @rdeodoo :smile:

Reply:
![image](https://github.com/odoo/odoo/assets/30048408/40f9ad62-fff8-4058-8437-dbe7385e96ea)